src/header/common/link.rs

```diff
@@ -0,0 +1,975 @@
+        self.media_type.as_ref()
+    }
+
+    /// Get the LinkValue's `link-extension` parameter
+    pub fn link_extension(&self) -> Option<&String> {
+        self.link_extension.as_ref()
+    }
+
+    /// Update LinkValue's `rel` parameter
+    pub fn push_rel(mut self, rel: RelationType) -> LinkValue {
+        let mut v = self.rel.take().unwrap_or(Vec::new());
+
+        v.push(rel);
+
+        self.rel = Some(v);
+
+        self
+    }
+
+    /// Set LinkValue's `anchor` parameter
+    pub fn set_anchor(mut self, anchor: &str) -> ::Result<LinkValue> {
+        match Uri::new(anchor) {
+            Err(_) => Err(::Error::Header),
+            Ok(uri) =>  {
+                self.anchor = Some(uri);
+
+                Ok(self)
+            }
+        }
+    }
+
+    /// Update LinkValue's `rev` parameter
+    pub fn push_rev(mut self, rev: RelationType) -> LinkValue {
+        let mut v = self.rev.take().unwrap_or(Vec::new());
+
+        v.push(rev);
+
+        self.rev = Some(v);
+
+        self
+    }
+
+    /// Update LinkValue's `hreflang` parameter
+    pub fn push_href_lang(mut self, language_tag: LanguageTag) -> LinkValue {
+        let mut v = self.href_lang.take().unwrap_or(Vec::new());
+
+        v.push(language_tag);
+
+        self.href_lang = Some(v);
+
+        self
+    }
+
+    /// Update LinkValue's `media` parameter
+    pub fn push_media_desc(mut self, media_desc: MediaDesc) -> LinkValue {
+        let mut v = self.media_desc.take().unwrap_or(Vec::new());
+
+        v.push(media_desc);
+
+        self.media_desc = Some(v);
+
+        self
+    }
+
+    /// Set LinkValue's `title` parameter
+    pub fn set_title(mut self, title: &str) -> LinkValue {
+        self.title = Some(String::from(title));
+
+        self
+    }
+
+    /// Set LinkValue's `title*` parameter
+    pub fn set_title_star(mut self, title_star: &str) -> LinkValue {
+        self.title_star = Some(String::from(title_star));
+
+        self
+    }
+
+    /// Set LinkValue's `type` parameter
+    pub fn set_media_type(mut self, media_type: &Mime) -> LinkValue {
+        self.media_type = Some(media_type.clone());
+
+        self
+    }
+
+    /// Set LinkValue's `link-extension` parameter
+    pub fn set_link_extension(mut self, link_extension: &str) -> LinkValue {
+        self.link_extension = Some(String::from(link_extension));
+
+        self
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Trait implementations
+////////////////////////////////////////////////////////////////////////////////
+
+impl Header for Link {
+    fn header_name() -> &'static str {
+        static NAME: &'static str = "Link";
+        NAME
+    }
+
+    fn parse_header(raw: &Raw) -> ::Result<Link> {
+        // TODO: This should probably change to support multiple link
+        //       headers in one request although we can have one link
+        //       header with multiple values.
+        from_one_raw_str(raw)
```

Comment:
I found in the RFC https://tools.ietf.org/html/rfc5988#section-5.5 that the `Link` header can contain multiple links in a single header, by being comma-separated, and so the normal header rules apply which state that multiple headers of the same name can be joined together.

As an example, taken from a few pieces in the RFC:

```
 Link: </TheBook/chapter2>; rel="previous", </TheBook/chapter4>; rel="next"
 Link: </TheBook/chapter3>; rel="canoncical"
```


So it seems this would need be able to accept several `Link`s, both if a line has them comma-separated, and if there are multiple lines.